src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -332,61 +332,57 @@ void IrcMessageHandler::addMessage(Communi::IrcMessage *_message,
 void IrcMessageHandler::handleRoomStateMessage(Communi::IrcMessage *message)
 {
     const auto &tags = message->tags();
-    auto app = getApp();
 
     // get twitch channel
     QString chanName;
     if (!trimChannelName(message->parameter(0), chanName))
     {
         return;
     }
-    auto chan = app->twitch.server->getChannelOrEmpty(chanName);
+    auto chan = getApp()->twitch.server->getChannelOrEmpty(chanName);
 
-    if (auto *twitchChannel = dynamic_cast<TwitchChannel *>(chan.get()))
+    auto *twitchChannel = dynamic_cast<TwitchChannel *>(chan.get());
+    if (!twitchChannel)
     {
-        // room-id
-        decltype(tags.find("xD")) it;
+        return;
+    }
 
-        if ((it = tags.find("room-id")) != tags.end())
-        {
-            auto roomId = it.value().toString();
+    // room-id
 
-            twitchChannel->setRoomId(roomId);
-        }
+    if (auto it = tags.find("room-id"); it != tags.end())
+    {
+        auto roomId = it.value().toString();
+        twitchChannel->setRoomId(roomId);
+    }
 
-        // Room modes
-        {
-            auto roomModes = *twitchChannel->accessRoomModes();
+    // Room modes
+    {
+        auto roomModes = *twitchChannel->accessRoomModes();
 
-            if ((it = tags.find("emote-only")) != tags.end())
-            {
-                roomModes.emoteOnly = it.value() == "1";
-            }
-            if ((it = tags.find("subs-only")) != tags.end())
-            {
-                roomModes.submode = it.value() == "1";
-            }
-            if ((it = tags.find("slow")) != tags.end())
-            {
-                roomModes.slowMode = it.value().toInt();
-            }
-            if ((it = tags.find("r9k")) != tags.end())
-            {
-                roomModes.r9k = it.value() == "1";
-            }
-            if ((it = tags.find("broadcaster-lang")) != tags.end())
-            {
-                roomModes.broadcasterLang = it.value().toString();
-            }
-            if ((it = tags.find("followers-only")) != tags.end())
-            {
-                roomModes.followerOnly = it.value().toInt();
-            }
-            twitchChannel->setRoomModes(roomModes);
+        if (auto it = tags.find("emote-only"); it != tags.end())
+        {
+            roomModes.emoteOnly = it.value() == "1";
         }
-
-        twitchChannel->roomModesChanged.invoke();
+        if (auto it = tags.find("subs-only"); it != tags.end())
+        {
+            roomModes.submode = it.value() == "1";
+        }
+        if (auto it = tags.find("slow"); it != tags.end())
+        {
+            roomModes.slowMode = it.value().toInt();
+        }
+        if (auto it = tags.find("r9k"); it != tags.end())
+        {
+            roomModes.r9k = it.value() == "1";
+        }
+        if (auto it = tags.find("followers-only"); it != tags.end())
+        {
+            roomModes.followerOnly = it.value().toInt();
+        }
```

Comment:
Rather than doing multiple iterations on a single iterator it would be nicer to iterate once and use a switch to apply these modes. I know it's not something you changed but could fit in this PR.

Also my knowledge of C++ iterators might be wrong since I'm thinking of them like an `IEnumerable<T>` from .NET

Reply:
Good point, I'll handle this in another PR though if needed as it's something that was here before and I didn't change this logic.